pkg/arvo/app/graph-store.hoon

```diff
@@ -412,90 +316,91 @@
-      ::
-      :-  (give [/updates]~ [%remove-nodes resource (~(uni in indices) affected-indices)])
+        (put:orm-log update-log time [time [%remove-posts resource indices]])
+      :-  (give [/updates]~ [%remove-posts resource indices])
       %_  state
-          update-logs  (~(put by update-logs) resource update-log)
+        update-logs  (~(put by update-logs) resource update-log)
+      ::
           graphs
         %+  ~(put by graphs)
           resource
-        [new-graph mark]
+        :_  mark
+        %^  remove-indices
+            resource
+          graph
+        (sort ~(tap in indices) by-lent)
       ==
       ::
-      :: we always want to remove the deepest node first,
-      :: so we don't remove parents before children
       ++  by-lent
         |*  [a=(list) b=(list)]
         ^-  ?
         (gth (lent a) (lent b))
       ::
       ++  remove-indices
-        =|  affected=(set index:store)
         |=  [=resource:store =graph:store indices=(list index:store)]
-        ^-  [(set index:store) graph:store]
-        ?~  indices  [affected graph]
-        =^  new-affected  graph
-          (remove-index graph i.indices)
+        ^-  graph:store
+        ?~  indices  graph
         %_  $
-            indices  t.indices
-            affected  (~(uni in affected) new-affected)
-        ==
-      ::
-      ++  get-descendants
-        |=  =graph:store
-        =|  indices=(list index:store)
-        =/  nodes  (tap:orm:store graph)
-        %-  ~(gas in *(set index:store))
-        |-  =*  tap-nodes  $
-        ^+  indices
-        %-  zing
-        %+  turn  nodes
-        |=  [atom =node:store]
-        ^-  (list index:store)
-        %+  welp
-          index.post.node^~
-        ?.  ?=(%graph -.children.node)
-          ~
-        %_  tap-nodes
-          nodes  (tap:orm p.children.node)
+          indices  t.indices
+          graph    (remove-index graph i.indices)
         ==
       ::
       ++  remove-index
-        =|  indices=(set index:store)
+        =|  parent-hash=(unit hash:store)
         |=  [=graph:store =index:store]
-        ^-  [(set index:store) graph:store]
-        ?~  index  [indices graph]
+        ^-  graph:store
+        ?~  index  graph
         =*  atom   i.index
+        %^  put:orm
+            graph
+          atom
         ::  last index in list
         ::
         ?~  t.index
-          =^  rm-node  graph  (del:orm graph atom)
-          ?~  rm-node  `graph
-          ?.  ?=(%graph -.children.u.rm-node)
-            `graph
-          =/  new-indices
-            (get-descendants p.children.u.rm-node)
-          [(~(uni in indices) new-indices) graph]
-        =/  =node:store
+          =/  =node:store
+            ~|  "cannot remove index that does not exist {<index>}"
+            (need (get:orm graph atom))
+          ~|  "cannot remove post that has already been removed"
+          ?>  ?=(%& -.post.node)
+          =*  p  p.post.node
+          %=    node
```

Comment:
`%_` is preferable here

Reply:
Not possible since it changes the type of maybe-post from the %y branch to the %n branch of the mint

Reply:
I have no object level opinion here, but it's worth noting that `&` or `|` don't have this problem -- their type is `?(%.y %.n)` (see also `.y` or `.n`, which i've never actually seen used ...).

Reply:
Oh you may have misunderstood.. this is just an each, so the actual values in question are %| and %&. I just colloquially referred to them as %y and %n (imprecise!)

Reply:
My bad, I was just skimming the comments. (If you *want* to use `%_`, you'll probably need to move `?> ?=(%& -.post.node)` inside it ...)

Reply:
Oh right, didn't catch the ?>

Reply:
Resolved to use the `?>` inside of a `%_` instead of using the `%=`. Thanks all!